lib/govuk_publishing_components/presenters/page_with_step_by_step_navigation.rb

```diff
@@ -81,12 +95,33 @@ def active_step_by_step
         @active_step_navs.first
       end
 
+      def secondary_step_by_steps
+        secondary_to_step_navs
+      end
+
+      def secondary_step_by_step?
+        secondary_step_by_steps.any?
+      end
+
+      def secondary_step_by_step
+        secondary_to_step_navs.first
+      end
+
+      def show_secondary_step_by_step?
+        step_navs_combined_list.count === 0 && secondary_step_by_step? && secondary_step_by_steps.count === 1
+      end
+
+      def show_related_links_for_secondary_step_by_steps?
+        step_navs_combined_list.count === 0 && secondary_step_by_step? && secondary_to_step_navs.count < 5
+      end
+
     private
 
       attr_reader :content_item, :current_path
 
       def current_step_nav
         return active_step_by_step if active_step_by_step?
+        return secondary_step_by_step if show_secondary_step_by_step?
```

Comment:
Also here, shouldn't the secondary step navs only be evaluated only if there aren't any primary step navs?

Reply:
as above - in the show_secondary_step_by_step we do a check to see if there are any step navs